client/src/entrypoints/admin/core.js

```diff
@@ -1,11 +1,13 @@
 import $ from 'jquery';
+import * as Stimulus from '@hotwired/stimulus';
 
 import { coreControllerDefinitions } from '../../controllers';
 import { escapeHtml } from '../../utils/text';
 import { initStimulus } from '../../includes/initStimulus';
 
 /** initialise Wagtail Stimulus application with core controller definitions */
-window.Stimulus = initStimulus({ definitions: coreControllerDefinitions });
+window.stimulusApp = initStimulus({ definitions: coreControllerDefinitions });
+window.Stimulus = Stimulus;
```

Comment:
I'm on board for

- `window.wagtail.app` for the Stimulus application instance. 
- `windoe.StimulusModule` for the module exposed export.

This avoids naming confusion or collisions with `window.Stimulus`/`window.stimulus`. Which means the devs will encounter this as the 'wrong' way much easier as it will be undefined. 

Let me know if you agree and I'm happy to make the change.

Reply:
After a bit of thinking, I feel that this is a much better approach.

We already have `window.wagtail.components` for our exposed React components, I think that this gives us a nicer way to expose globals long term if needed.

The `StimulusModule` on it's own is a good approach, maybe we can remove this one day and provide a nicer module import approach (if we can make this work easily with Django static JS approaches).

I will update this PR accordingly.

I have a backup of the current approach if needed though.

Reply:
Ok. This is done. I've also moved the Wagtail object stuff into the one file, added unit tests for how we set up.this object and added JSDoc. 

Happy to move this all to wagtailadmin.js instead but maybe that's best done as a seperate issue when a few more things are migrated away from core/Wagtail admin.js

Reply:
@thibaudcolas  when you have a min, could you look overt this PR again and let me know if we are good to go with these last set of naming changes applied.